mesonbuild/optinterpreter.py

```diff
@@ -49,18 +72,20 @@ class OptionException(mesonlib.MesonException):
 
 optname_regex = re.compile('[^a-zA-Z0-9_-]')
 
+_YIELD_KW: KwargInfo[bool] = KwargInfo('yield', bool, default=coredata.default_yielding, since='0.45.0')
```

Comment:
Hmm, come to think of it maybe we should pass this in at the global option method and typecheck it there, then pass it outside the kwargs dict to the individual parsers? It's a universal option.

description is also a universal option, and that's how we do it.
deprecated is also a universal option, but that's not part of a UserOption constructor, not completely sure why the only way to set it is via `opt.deprecated = ...`

Reply:
that, it turns out, causes issue for the `typed_kwargs()` function, which assumes a function with 3 or 4 parameters is in the last (4th) position. I think it really is easier to just leave it in the kwargs and let the individual parsers handle it.

Reply:
Right, fair enough.